ls1intum/Artemis
# [Bugfix] Fix Issues for preliminary results

### Checklist


### Motivation and Context


Resolves #3333

### Description

This PR adds the "(preliminary)" text for results that are preliminary. Furthermore, preliminary results are marked in a grey color and have a question mark icon.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Generate a Java Programming Exercise. Make sure you set the dates to the future (end date and assessment dates) and select the static analysis tool with a 0% penalty
4. Click Edit in Editor and then click on Create Assignment Repository
5. Navigate to the assignment repository, edit BubbleSort.java and add `Collections.sort(input);`, and submit the code
6. The result should be colored in grey and include the "(preliminary)" text
7. Re-add the semicolon, submit the code, and navigate to "Grading".
8. Set the visibility of each test to "after due date", press "Save", and "Trigger All".
9. Navigate back to the code editor, select the assignment, and make sure the text is colored in grey and includes the "preliminary" string 
10. Click on the result and make sure the modal displayed below opens up.

### Test Coverage




### Screenshots


The visibility of all tests are set to "after due date" and the build compiles:
![image](https://user-images.githubusercontent.com/11298674/121855299-76f7c280-ccf3-11eb-98e8-893ab8a7d268.png)

Clicking on the preliminary result (all tests are marked as "after due date") opens up a modal dialog:
![image](https://user-images.githubusercontent.com/11298674/121857674-29c92000-ccf6-11eb-92df-e7b33ee92d10.png)

The visibility of a single test is set to "Always" and the code implements it:
![image](https://user-images.githubusercontent.com/11298674/121855574-cd650100-ccf3-11eb-8924-29a9b222e04a.png)

If the build fails, the text is red instead of grey:
![image](https://user-images.githubusercontent.com/11298674/121855820-10bf6f80-ccf4-11eb-8668-e2144bf96113.png)
